spring-boot-project/spring-boot-actuator-autoconfigure/src/test/java/org/springframework/boot/actuate/autoconfigure/observation/ObservationAutoConfigurationTests.java

```diff
@@ -129,18 +151,35 @@ void autoConfiguresObservationHandlerWithCustomContext() {
 					ObservationRegistry observationRegistry = context.getBean(ObservationRegistry.class);
 					List<ObservationHandler<?>> handlers = context.getBean(CalledHandlers.class).getCalledHandlers();
 					CustomContext customContext = new CustomContext();
-					Observation.start("test-observation", () -> customContext, observationRegistry);
+					Observation.start("test-observation", () -> customContext, observationRegistry).stop();
 					assertThat(handlers).hasSize(1);
 					assertThat(handlers.get(0)).isInstanceOf(ObservationHandlerWithCustomContext.class);
+					assertThat(context).hasSingleBean(DefaultMeterObservationHandler.class);
+					assertThat(context).doesNotHaveBean(TracingAwareMeterObservationHandler.class);
 				});
 	}
 
+	@Test
+	void autoConfiguresTracingAwareMeterObservationHandler() {
+		this.tracingContextRunner.withUserConfiguration(CustomTracingObservationHandlers.class).run((context) -> {
+			ObservationRegistry observationRegistry = context.getBean(ObservationRegistry.class);
+			List<ObservationHandler<?>> handlers = context.getBean(CalledHandlers.class).getCalledHandlers();
+			// Intentionally not stopped since that will trigger additional logic in
+			// TracingAwareMeterObservationHandler that we don't test here
+			Observation.start("test-observation", observationRegistry);
+			assertThat(handlers).hasSize(1);
+			assertThat(handlers.get(0)).isInstanceOf(CustomTracingObservationHandler.class);
+			assertThat(context).hasSingleBean(TracingAwareMeterObservationHandler.class);
+			assertThat(context.getBeansOfType(ObservationHandler.class)).hasSize(2);
+		});
+	}
+
 	@Test
 	void autoConfiguresObservationHandlerWhenTracingIsActive() {
 		this.tracingContextRunner.withUserConfiguration(ObservationHandlersTracing.class).run((context) -> {
 			ObservationRegistry observationRegistry = context.getBean(ObservationRegistry.class);
 			List<ObservationHandler<?>> handlers = context.getBean(CalledHandlers.class).getCalledHandlers();
-			Observation.start("test-observation", observationRegistry);
+			Observation.start("test-observation", observationRegistry).stop();
```

Comment:
As far as I can tell, this change is unrelated to wrapping the `DefaultMeterObservationHandler` in a `TracingAwareMeterObservationHandler`. If I've got that right, it should be handled separately please.